### PR TITLE
(fix)RadioGroup: Size prop mismatch, configure style

### PR DESCRIPTION
**Pull request by cehsu**
These changes fix a bug where `<RadioGroup/>` accepted a `size` (string) prop, but it's child component `<Checkbox/>` only accepted `iconSize` (number), and `fontSize` (number). Also styles are now passed down to support more customization (margins, etc.), such as required by [EMR Message Conversation List](https://app.asana.com/0/821652619613450/1108680987294666/f).

image

### CODE REVIEW

File: src/Form/Checkbox.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ const CheckboxContainer = createComponent({
       margin-left: ${p => (p.horizontal ? '12px' : 0)};
       margin-top: ${p => (p.horizontal ? 0 : '4px')};
     }
+    ${p => p.style}
```
Comment (cehsu) [inline src/Form/Checkbox.js]: Given how the margins are set above, this seemed like the least ornate (e.g. specifying a `margin` prop, which I considered earlier), but still effective option. Thanks for any thoughts on whether this isn't intuitive or consistent.

Comment (choochootrain) [reply src/Form/Checkbox.js]: for those (like me) who dont understand this dark magic:
https://github.com/styled-components/styled-components/blob/master/packages/styled-components/src/constructors/css.js
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates

Comment (kylealwyn) [reply src/Form/Checkbox.js]: I don't think line is necessary. styled-components take a style prop by default

File: src/Form/RadioGroup.js [hunk 2]
```diff
@@ -60,7 +63,7 @@ class RadioGroup extends Component {
   };
 
   render() {
-    const { choices, error, horizontal, label, name, colorOn, colorOff, size } = this.props;
+    const { choices, error, horizontal, label, name, colorOn, colorOff, fontSize, iconSize, styles } = this.props;
```
Comment (kylealwyn) [inline src/Form/RadioGroup.js]: we could spread these final five props into something like `...checkboxProps` and spread onto `<Checkbox />` below

File: src/Form/RadioGroup.js [hunk 3]
```diff
@@ -87,7 +90,9 @@ class RadioGroup extends Component {
                   valueFalse={value}
                   iconOn="radiobox-marked"
```
Comment (kylealwyn) [inline src/Form/RadioGroup.js]: Can we add `iconOn` and `iconOff` as props while we're here and set defaultProps to these values?

File: src/Form/Checkbox.js (hunk 1 above)
Comment (cehsu) [reply src/Form/Checkbox.js]: Hm, I tried removing this line and the style no longer got applied.

File: src/Form/RadioGroup.js (hunk 2 above)
Comment (cehsu) [reply src/Form/RadioGroup.js]: Sounds good, will update.

File: src/Form/RadioGroup.js (hunk 3 above)
Comment (cehsu) [reply src/Form/RadioGroup.js]: Sounds good, will update.

File: src/Form/RadioGroup.js (hunk 2 above)
Comment (cehsu) [reply src/Form/RadioGroup.js]: Thanks, updated.

File: src/Form/RadioGroup.js (hunk 3 above)
Comment (cehsu) [reply src/Form/RadioGroup.js]: Thanks, updated.

File: src/Form/Checkbox.js (hunk 1 above)
Comment (kylealwyn) [reply src/Form/Checkbox.js]: It looks like it's an issue with the test style in the doc. If you exchange `"margin: 0 !important"` for 0, it works without this line

File: src/Form/RadioGroup.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { css } from 'styled-components';
```
Comment (kylealwyn) [inline src/Form/RadioGroup.js]: unused

File: src/Form/RadioGroup.js [hunk 5]
```diff
@@ -20,20 +21,26 @@ class RadioGroup extends Component {
...
     choices: PropTypes.arrayOf(
       PropTypes.shape({
         value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
         label: PropTypes.string,
         disabled: PropTypes.bool,
       })
     ),
+    styles: PropTypes.object,
+    iconOn: PropTypes.string,
+    iconOf: PropTypes.string,
```
Comment (kylealwyn) [inline src/Form/RadioGroup.js]: typo

File: src/Form/RadioGroup.js [hunk 6]
```diff
@@ -20,20 +21,26 @@ class RadioGroup extends Component {
...
+    fontSize: PropTypes.number,
+    iconSize: PropTypes.number,
     choices: PropTypes.arrayOf(
       PropTypes.shape({
         value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
         label: PropTypes.string,
         disabled: PropTypes.bool,
       })
     ),
+    styles: PropTypes.object,
```
Comment (kylealwyn) [inline src/Form/RadioGroup.js]: prefer shape() to make linter happy

File: src/Form/RadioGroup.js [hunk 7]
```diff
@@ -60,7 +67,7 @@ class RadioGroup extends Component {
   };
 
   render() {
-    const { choices, error, horizontal, label, name, colorOn, colorOff, size } = this.props;
+    const { choices, error, horizontal, label, name, colorOn, colorOff, ...checkboxProps } = this.props;
```
Comment (kylealwyn) [inline src/Form/RadioGroup.js]: colorOn and colorOff and be swept into checkboxProps spread

File: src/Form/Checkbox.js [hunk 8]
```diff
@@ -141,7 +145,7 @@ class Checkbox extends React.Component {
         <Flex alignItems="center">
           <StyledIcon size={iconSize} color={checked ? colorOn : colorOff} name={checked ? iconOn : iconOff} />
 
-          {label && <StyledLabel fontSize={fontSize}>{label}</StyledLabel>}
+          {label && <StyledLabel fontSize={fontSize} style={styles.Label}>{label}</StyledLabel>}
```
Comment (cehsu) [inline src/Form/Checkbox.js]: This style is applied without specifying it in the styled component's template literal, I'm not sure why the other style does not.

Comment (kylealwyn) [reply src/Form/Checkbox.js]: https://github.com/heydoctor/molekule/pull/28/files#r266171170

File: src/Form/Checkbox.js (hunk 1 above)
Comment (cehsu) [reply src/Form/Checkbox.js]: Oh thanks, I'll do that.

File: src/Form/RadioGroup.js (hunk 6 above)
Comment (cehsu) [reply src/Form/RadioGroup.js]: Thanks, replaced with `shape()`.

File: src/Form/RadioGroup.js (hunk 7 above)
Comment (cehsu) [reply src/Form/RadioGroup.js]: OK, removed.

File: src/Form/RadioGroup.js (hunk 5 above)
Comment (cehsu) [reply src/Form/RadioGroup.js]: Thanks, fixed.

File: src/Form/RadioGroup.js (hunk 4 above)
Comment (cehsu) [reply src/Form/RadioGroup.js]: Thanks, removed.

**Pull request merged**